app/models/option_set_import.rb

```diff
@@ -105,8 +107,9 @@ def create_option_set
         end
       end
 
-      raise ActiveRecord::Rollback if errors.present?
+      @option_set.generate_sequence(batch: true)
```

Comment:
I can see why you'd want to do it this way as it reduces duplication, but I think it would be much more efficient to assign the sequence during the initial creation of each option node above. Option Sets can have 100,000's of options, so need to be super performance conscious here.